### ISSUE TITLE
No nonlinear layers

### ISSUE THREAD

**Issue by sebafudi** [Zip-o-mat/Slic3r-with-Electronics#4]
### Version
#1 

### Operating system type + version
Ubuntu 18.04 LTS

### Behavior
Slic3r won't generate nonlinear layers. In console it says, that there's a collision. I've tried to use my printer and filament settings, but it was the same thing. Attached config is mostly default with only nonlinear settings changed.
![vmconnect_2019-08-27_13-45-40](https://user-images.githubusercontent.com/5780459/63768972-e3f95880-c8d1-11e9-9628-3d09342f5d9c.png)

#### STL/Config (.ZIP) where problem occurs
[stl_config.zip](https://github.com/Zip-o-mat/Slic3r/files/3545931/stl_config.zip)

**Comment by platsch** [comment]
Yes, that probably is correct behavior, the model you are trying to slice has no surface which could be printed nonplanar without colliding with some other part of the model.
The collision checking is a tricky thing, it often finds collisions that are hard to spot for a human. That can probably be improved to allow for some tolerances or subdivide the object into areas which can still partly be printed nonplanar, but someone has to take the effort :-)
A visualization of collision areas would also be neat.

**Comment by Zip-o-mat** [comment]
Maybe a better visualisation would help to see why there are no nonplanar surfaces generated.

**Comment by ds-motivo** [comment]
(EDIT) See next post. Everything is working as expected.

I'm also _not_ getting any nonplanar layers, no matter my settings. Here's a ball (half of a pokeball) I got from thingiverse. I've also tried a few topological map models as well.

![image](https://user-images.githubusercontent.com/25038705/64298674-433b1680-cf2b-11e9-8c99-07e62e6282d0.png)

My nonplanar settings are 45 degrees nozzle and 25mm of clearance (just to test, not actually my printer settings).

**Comment by ds-motivo** [comment]
OK I take back my previous post. I guess I'm not intuitively understanding the role of nozzle clearance height in this. Changing just a couple millimeters drastically affects your "reach" on certain geometries.

**Comment by platsch** [comment]
Intuitively I would reckon 45° and 25mm should be sufficient to get at least some non-planar coverage.
My first guess for an explanation would be that slic3r only uses full facets and depending on the tesselation of your model it is possible that some facets span a higher distance than 25mm, preventing detection of a non-planar region.
Also: the error reports a collision...
@Zip-o-mat do you have an explanation?

**Comment by Zip-o-mat** [comment]
First of all there is no existing printer(at least not that I know) with 45° and 25mm clearance.
There is a collision detected but it seems that this is not correct. I have to check the collision detection.
@ds-motivo can you upload the stl or post a link?

**Comment by ds-motivo** [comment]
STLs attached below (in ZIPs). 

I only put the settings to 45deg/25mm because I figured it MUST produce some results. Couldn't make it happen on the pokeball, but _could_ make it happen on the cat toy.

I also have a file from a customer that I can't share that is a horn cover (the thing you press on your car's steering wheel), shallow top surface but couldn't get nonplanar layers with any settings.

[Cattoy1.zip](https://github.com/Zip-o-mat/Slic3r/files/3585794/Cattoy1.zip)

[pokeball_side.zip](https://github.com/Zip-o-mat/Slic3r/files/3585741/pokeball_side.zip)

![image](https://user-images.githubusercontent.com/25038705/64462096-2847de00-d0b4-11e9-995d-4df29a43b7b9.png)

**Comment by wookie76** [comment]
Greetings,

I too was playing around with silly settings just to see how far it could go.  Is there a "soft" limit of 10mm clearance and 50°?  It didn't seem to change much beyond that.  It didn't seem to like straying too far from any of the default settings.

Thanks for the hard work,
Bryce

**Comment by Zip-o-mat** [comment]
There should be no limits but at the moment the collision detection is rather sensitive and has many false positive collisions. I am working on the next version with better collision detection in nonplanar toolpaths.

**Comment by wookie76** [comment]
That's great.  And again this is just nit-picking.  Overall it's quite
impressive what you've accomplished, thanks again.

**Comment by ds-motivo** [comment]
Agreed! Keep at it, with enough traction I think this could influence the way 3d printers are designed. I'm already seeing a ton of the 3d printing youtubers take a crack at it.
